src/lib.rs

```diff
@@ -316,6 +356,38 @@ impl Member {
         quote!()
     }
 
+    /// Returns (format_string, argument) tuple.
+    fn defmt(&self) -> Option<(String, TokenStream)> {
+        let inner = self.inner.as_ref()?;
+
+        if inner.from.is_empty() {
+            return None;
+        }
+
+        // default to using {:?}
+        let mut spec = "{:?}".to_owned();
+
+        // primitives supported by defmt
+        const PRIMITIVES: &[&str] = &[
+            "bool", "usize", "isize", "u8", "u16", "u32", "u64", "u128", "i8", "i16", "i32", "i64",
+            "i128", "f32", "f64",
+        ];
+
+        // get the type name so we can use more efficient defmt formats
+        // if it's a primitive
+        if let syn::Type::Path(syn::TypePath { ref path, .. }) = inner.ty {
+            if let Some(ident) = path.get_ident() {
+                if PRIMITIVES.iter().any(|s| ident == s) {
+                    // defmt supports this primitive, use special spec
+                    spec = format!("{{={}}}", ident);
+                }
+            }
+        }
+
+        let ident = &inner.ident;
+        Some((format!("{}: {}", ident, spec), quote!(self.#ident())))
+    }
```

Comment:
If you use plain `format!`, inlining the variables is a bit more readable. (Only `format_ident!` from quote breaks when you do this.)
```suggestion
        Some((format!("{ident}: {spec}"), quote!(self.#ident())))
```

Reply:
Well, the suggestions are on the wrong line, sorry.